ruma-federation-api/src/membership/create_invite/v2.rs

```diff
@@ -0,0 +1,90 @@
+//! [PUT /_matrix/federation/v2/invite/{roomId}/{eventId}](https://matrix.org/docs/spec/server_server/r0.1.4#put-matrix-federation-v2-invite-roomid-eventid)
+
+use js_int::UInt;
+use ruma_api::ruma_api;
+use ruma_events::{room::member::MemberEventContent, EventType};
+use ruma_identifiers::{EventId, RoomId, RoomVersionId, ServerName, UserId};
+
+use super::{InviteEvent, StrippedState};
+
+ruma_api! {
+    metadata: {
+        description: "Invites a remote user to a room.",
+        method: PUT,
+        name: "create_invite",
+        path: "/_matrix/federation/v2/invite/:room_id/:event_id",
+        rate_limited: false,
+        requires_authentication: true,
+    }
+
+    request: {
+        /// The room ID that the user is being invited to.
+        #[ruma_api(path)]
+        room_id: RoomId,
+
+        /// The event ID for the invite event, generated by the inviting server.
+        #[ruma_api(path)]
+        event_id: EventId,
+
+        /// The version of the room where the user is being invited to.
+        room_version: RoomVersionId,
+
+        /// An invite event.
+        event: InviteEvent,
+
+        /// An optional list of simplified events to help the receiver of the invite identify the room.
+        invite_room_state: StrippedState,
+    }
+
+    response: {
+        /// An invite event.
+        event: InviteEvent,
+    }
+}
+
+/// Initial set of fields for `Response`.
+pub struct ResponseInit {
+    /// The matrix ID of the user who sent the original `m.room.third_party_invite`.
+    sender: UserId,
+
+    /// The name of the inviting homeserver.
+    origin: Box<ServerName>,
+
+    /// A timestamp added by the inviting homeserver.
+    origin_server_ts: UInt,
+
+    /// The user ID of the invited member.
+    state_key: UserId,
+
+    /// The content of the event. Must include a `membership` of invite.
+    content: MemberEventContent,
+}
+
+impl Request {
+    /// Creates a new `Request` with the given parameters
+    pub fn new(
+        room_id: RoomId,
+        event_id: EventId,
+        room_version: RoomVersionId,
+        event: InviteEvent,
+        invite_room_state: StrippedState,
+    ) -> Self {
+        Self { room_id, event_id, room_version, event, invite_room_state }
+    }
+}
+
+impl Response {
```

Comment:
`Response` should have a regular constructor since there is only one field to initialize. This code can still be useful though as `impl From<InviteEventInit> for InviteEvent` – `InviteEvent` should also be `#[non_exhaustive]` as well as the request and response blocks, these accept attributes now; see also the new section of CONTRIBUTING.md: [(Type) Privacy and Forwards Compatibility](https://github.com/ruma/ruma/blob/master/CONTRIBUTING.md#type-privacy-and-forwards-compatiblity).